Bug1270087 - Auto fit password bullets for landscape view

tylerlacroix opened the pull request.

sleroux commented on Client/Frontend/AuthenticationManager/PasscodeViews.swift:
```diff
@@ -162,6 +163,9 @@ class PasscodePane: UIView {
         }
+        
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(PasscodePane.keyboardWillHide(_:)), name: UIKeyboardWillHideNotification, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(PasscodePane.keyboardWillShow(_:)), name: UIKeyboardWillShowNotification, object: nil)
```
Whenever registering observers in `init`, you'll want to pair it up with a `removeObserver` call in the `deinit` method so it doesn't linger after the VC is gone.

sleroux commented on Client/Frontend/AuthenticationManager/PasscodeViews.swift:
```diff
@@ -179,4 +183,19 @@ class PasscodePane: UIView {
     }
+    
+    func keyboardWillShow(sender: NSNotification) {
+        let keyboardFrame: CGRect = sender.userInfo![UIKeyboardFrameEndUserInfoKey]!.CGRectValue
```
Nit: As a general tip, try to avoid using force unwraps if you can since they can often lead to runtime exceptions. I would try writing this using guard and using optionals instead so it's safer.

```swift
guard let keyboardFrame = sender.userInfo?[UIKeyboardFrameEndUserInfoKey]?.CGRectValue else {
    return
}
```

sleroux commented:
When turning on slow mo animations, I notice that the passcode pane view 'scales' in. Looks like the animation from the keyboard observer is firing and the frames aren't setup yet which causes the animation to scale from a frame of 0.

![passcode_animation_issue](https://cloud.githubusercontent.com/assets/434322/15183999/850dc060-1761-11e6-9fcd-62bada0c017d.gif)

I might be better to add/remove the observers in `viewDidAppear` and `viewWillDisappear` instead.

tylerlacroix commented:
Thanks Steph! I added the deinit, optional unwrapping, and I added a layoutIfNeeded() call to fix the animation because the PasscodePane is a UIView so it doesn't have viewDidAppear or viewWillDisappear.
![may11](https://cloud.githubusercontent.com/assets/11801330/15187534/7152f884-176f-11e6-9eb5-c75496144b74.gif)
